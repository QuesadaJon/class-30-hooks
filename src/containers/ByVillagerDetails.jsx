import React, { useEffect, useState } from 'react';
import { useParams, Link } from 'react-router-dom';
import { findDetailsByVillager } from '../services/animalCrossingApi';

export default function ByVillagerDetails() {
  const [villager, setVillager] = useState({});
  const { id } = useParams();

  useEffect(() => {
    findDetailsByVillager(id)
      .then((villager) => setVillager(villager));
  }, []);

  const { name, image, quote, personality, species, birthday } = villager;

  return (
    <div>
      <h1>{name}</h1>
      <img src={image}/>
      <figcaption>
        <p>{quote}</p>
        <p>{personality}</p>
        <p>{species}</p>
        <p>{birthday}</p>
      </figcaption>
      <Link to="/">List Page</Link>
    </div>
  );
}
