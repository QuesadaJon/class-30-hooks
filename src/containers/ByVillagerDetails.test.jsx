import React from 'react';
import { render } from '@testing-library/react';
import ByVillagerDetails from './ByVillagerDetails';
import { Router } from 'react-router-dom';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: jest.fn().mockReturnValue({ _id: '5' }),
}));

describe('ByVillagerDetails component', () => {
  it('matches a snapshot', () => {
    const { asFragment } = render(
      <Router>
        <ByVillagerDetails />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
