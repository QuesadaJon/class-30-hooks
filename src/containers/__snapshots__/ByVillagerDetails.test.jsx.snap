// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ByVillagerDetails component matches a snapshot 1`] = `
<DocumentFragment>
  <div
    class="VillagerDetails"
  >
    <h1 />
    <img />
    <figcaption>
      <p />
      <p />
      <p />
      <p />
    </figcaption>
    <a
      href="/"
    >
      List Page
    </a>
  </div>
</DocumentFragment>
`;
